src/Host/Client/Impl/Host/RemoteBrokerClient.cs

```diff
@@ -55,26 +64,33 @@ public RemoteBrokerClient(string name, Uri brokerUri, ICoreServices services, IC
             if (sslPolicyErrors.HasFlag(SslPolicyErrors.RemoteCertificateNotAvailable)) {
                 Log.WriteAsync(LogVerbosity.Minimal, MessageCategory.Error, Resources.Error_NoBrokerCertificate).DoNotWait();
                 _console.Write(Resources.Error_NoBrokerCertificate);
-            } else {
-                var hashString = certificate.GetCertHashString();
-                bool accepted = true;
+                return false;
+            }
 
+            _verificationSemaphore.Wait();
```

Comment:
I think simple `lock` should be enough here (since we don't have any async code).